packages/shared/apis/host/SignedFetch.ts

```diff
@@ -0,0 +1,56 @@
+import { signedFetch } from './../../../atomicHelpers/signedFetch'
+import { ETHEREUM_NETWORK } from './../../../config'
+import { getSelectedNetwork } from './../../dao/selectors'
+import { store } from './../../store/isolatedStore'
+import { getIsGuestLogin } from './../../session/selectors'
+import { onLoginCompleted } from './../../session/sagas'
+import { getRealm } from './../../comms/selectors'
+
+import { RpcServerPort } from '@dcl/rpc'
+import { PortContext } from './context'
+import * as codegen from '@dcl/rpc/dist/codegen'
+
+import { SignedFetchServiceDefinition } from '../proto/SignedFetch.gen'
+
+export function registerSignedFetchServiceServerImplementation(port: RpcServerPort<PortContext>) {
+  codegen.registerService(port, SignedFetchServiceDefinition, async () => ({
+    async signedFetch(req, ctx) {
+      const { identity } = await onLoginCompleted()
+
+      const state = store.getState()
+      const realm = getRealm(state)
+      const isGuest = !!getIsGuestLogin(state)
+      const network = getSelectedNetwork(state)
+
+      const compatibilityRealm:
+        | {
+            domain: string
+            layer: string
+            catalystName: string
+          }
```

Comment:
where does this type comes from?

Reply:
https://github.com/decentraland/kernel/blob/main/packages/shared/apis/SignedFetch.ts#L26